src/validation.cpp

```diff
@@ -954,8 +958,27 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
     }
 
     ws.m_ancestors = *ancestors;
-    if (const auto err_string{SingleV3Checks(ws.m_ptx, ws.m_ancestors, ws.m_conflicts, ws.m_vsize)}) {
-        return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "v3-rule-violation", *err_string);
+    // Even though just checking direct mempool parents for inheritance would be sufficient, we
+    // check using the full ancestor set here because it's more convenient to use what we have
+    // already calculated.
+    if (const auto err{SingleV3Checks(ws.m_ptx, ws.m_ancestors, ws.m_conflicts, ws.m_vsize)}) {
+        // Disabled within package validation.
+        if (err->second != nullptr && !args.m_package_submission && args.m_allow_replacement) {
+            // Potential sibling eviction. Add the sibling to our list of mempool conflicts to be
+            // included in RBF checks.
+            ws.m_conflicts.insert(err->second->GetHash());
```

Comment:
Seems like the documentation for `m_conflicts` and `m_iters_conflicting` should change. They're currently documented as `/** Txids of mempool transactions that this transaction directly conflicts with. */
`. Is that still accurate if we add transactions that don't conflict due to spending the same outputs?

Reply:
We're treating it as a "direct conflict" for RBF reasons, although indeed maybe the doc should get updated to mention the sibling eviction case?